java-bootcamp-2018-tdd/src/main/java/com/globant/business/shoppingCart/ShoppingCart.java

```diff
@@ -0,0 +1,99 @@
+package com.globant.business.shoppingCart;
+
+import java.util.Collection;
+import java.util.Hashtable;
+
+public class ShoppingCart implements IShoppingCart {
+	/**
+	 * Hash whit the rows
+	 */
+	Hashtable<String, RowCart> products;
+
+	/**
+	 * constructor
+	 */
+	public ShoppingCart() {
+		products = new Hashtable<String, RowCart>();
+	}
+
+	/**
+	 * add a new product to the cart whit its quantity if the product is already in
+	 * the cart add the amount
+	 */
+	public Product addProduct(String name, String category, float price, int quantity) {
+		if (!products.containsKey(name)) {
+			RowCart row = new RowCart(new Product(name, category, price), quantity);
+			products.put(name, row);
+		} else {
+			products.get(name).addQuantity(quantity);
+		}
+		return products.get(name).getProduct();
+	}
+
+	/**
+	 * set the amount of one product in the cart if the new quantity is less than 1,
+	 * remove the product
+	 */
+	public Product changeamount(String product, int quantity) {
+		if (quantity > 0) {
+			RowCart current = products.get(product);
+			current.changeAmmount(quantity);
+			return current.getProduct();
+		} else {
+			RowCart current = products.get(product);

```

Comment:
it the product does not exist, this will raise an error. also, you are not checking the quantity on the current cart for removing, you are just removing.

Reply:
In this method, the new quantity is set, regardless of the amount existing before. I can change it for add a quantity (positive or negative), and that add with the existing